c10/cuda/CUDACachingAllocator.cpp

```diff
@@ -1504,5 +1504,13 @@ void raw_delete(void* ptr) {
 
 } // namespace CUDACachingAllocator
 
+std::pair<size_t, size_t> raw_cuda_mem_get_info(int device) {
+  C10_CUDA_CHECK(cudaGetDevice(&device));
+  size_t device_free;
+  size_t device_total;
+  C10_CUDA_CHECK(cudaMemGetInfo(&device_free, &device_total));
+  return {device_free, device_total};
+}
```

Comment:
nit: There isn't really any reason to put this in CUDACachingAllocator.cpp, it doesn't need any access to the internal allocator, and really doesn't have anything to do with the caching allocator at all

Reply:
Good point! I have re-defined this logic in `torch/csrc/cuda/shared/cudart.cpp`, as suggested. Please see: [https://github.com/pytorch/pytorch/pull/58635/commits/7d5c98a0306f23ad438302f29ab8fa321c316bcb](https://github.com/pytorch/pytorch/pull/58635/commits/7d5c98a0306f23ad438302f29ab8fa321c316bcb).